csrc/ops/arith.cpp

```diff
@@ -2283,6 +2283,103 @@ TensorView* argsort(
   return out->as<TensorView>();
 }
 
+namespace {
+// Create output tensor for grouped matrix multiplication
+// For grouped MM, determine output shape based on mat1 and mat2 structures.
+//
+// case 1:
+//   mat1   [m, k]
+//   mat2   [k, n]
+//   offset [g]
+//   output -> [g, m, n]
+// case 2:
+//   mat1   [g, m, k]
+//   mat2   [k, n]
+//   offset [g]
+//   output -> [m, n]
+// case 3:
+//   mat1   [m, k]
+//   mat2   [g, k, n]
+//   offset [g]
+//   output -> [m, n]
+TensorView* create_grouped_mm_output(
+    TensorView* mat1,
+    TensorView* mat2,
+    TensorView* offsets,
+    std::optional<DataType> dtype) {
+  const auto mat1_domain = TensorDomain::noReductions(mat1->getLogicalDomain());
+  const auto mat2_domain = TensorDomain::noReductions(mat2->getLogicalDomain());
+  const auto offs_domain =
+      TensorDomain::noReductions(offsets->getLogicalDomain());
+
+  NVF_CHECK(offs_domain.size() == 1, "offsets needs to be 1-D for grouped mm");
+
+  std::vector<IterDomain*> out_domain;
+
+  if (mat1->nDims() == 2 && mat2->nDims() == 2) {
```

Comment:
```suggestion
  if (mat1_domain.size() == 2 && mat2_domain.size() == 2) {
```
nDims includes reduction dimensions and will also be a problem for DID loop splits.